server/src/main/java/org/opensearch/cluster/metadata/DecommissionAttributeMetadata.java

```diff
@@ -0,0 +1,228 @@
+
+        if (!status.equals(that.status)) return false;
+        return decommissionAttribute.equals(that.decommissionAttribute);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(attributeType, decommissionAttribute, status);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getWriteableName() {
+        return TYPE;
+    }
+
+    @Override
+    public Version getMinimalSupportedVersion() {
+        return Version.V_2_3_0;
+    }
+
+    public DecommissionAttributeMetadata(StreamInput in) throws IOException {
+        this.decommissionAttribute = new DecommissionAttribute(in);
+        this.status = DecommissionStatus.fromString(in.readString());
+    }
+
+    public static NamedDiff<Custom> readDiffFrom(StreamInput in) throws IOException {
+        return readDiffFrom(Custom.class, TYPE, in);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        decommissionAttribute.writeTo(out);
+        out.writeString(status.status());
+    }
+
+    public static DecommissionAttributeMetadata fromXContent(XContentParser parser) throws IOException {
+        XContentParser.Token token;
+        DecommissionAttribute decommissionAttribute = null;
+        DecommissionStatus status = null;
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                String currentFieldName = parser.currentName();
+                if (attributeType.equals(currentFieldName)) {
+                    if (parser.nextToken() != XContentParser.Token.START_OBJECT) {
+                        throw new OpenSearchParseException(
+                            "failed to parse decommission attribute type [{}], expected object",
+                            attributeType
+                        );
+                    }
+                    token = parser.nextToken();
+                    if (token != XContentParser.Token.END_OBJECT) {
+                        if (token == XContentParser.Token.FIELD_NAME) {
+                            String fieldName = parser.currentName();
+                            String value;
+                            token = parser.nextToken();
+                            if (token == XContentParser.Token.VALUE_STRING) {
+                                value = parser.text();
+                            } else {
+                                throw new OpenSearchParseException(
+                                    "failed to parse attribute [{}], expected string for attribute value",
+                                    fieldName
+                                );
+                            }
+                            decommissionAttribute = new DecommissionAttribute(fieldName, value);
+                            token = parser.nextToken();
```

Comment:
whats the use of this line. We are already doing next token in while loop. Are we skipping this token. 
If yes then no need to assign to token variable. We can just do parser.nextoken()